### PR TITLE
✨ Add an admin dashboard

### PR DESCRIPTION
**Pull request by Esseme**
We have added an admin dashboard so that the admin user can easily
moderate the site.

### CODE REVIEW

**Comment by Esseme** [comment]
[Add an admin panel for the admin user to access and be able to moderate the site/app](https://trello.com/c/bnD9nYkl/79-add-an-admin-panel-for-the-admin-user-to-access-and-be-able-to-moderate-the-site-app)

File: config/routes.rb [hunk 1]
```diff
@@ -1,4 +1,10 @@
 Rails.application.routes.draw do
+  namespace :admin do
+    resources :articles
+    resources :users
+
+    root to: "articles#index"
```
Comment (gkosmo) [inline config/routes.rb:6]: I'm not sure we need to redefine the `root` :thinking:

File: app/dashboards/article_dashboard.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+class ArticleDashboard < Administrate::BaseDashboard
+  # ATTRIBUTE_TYPES
+  # a hash that describes the type of each of the model's fields.
+  #
+  # Each different type represents an Administrate::Field object,
+  # which determines how the attribute is displayed
+  # on pages throughout the dashboard.
+  ATTRIBUTE_TYPES = {
+    id: Field::Number,
```
Comment (gkosmo) [inline app/dashboards/article_dashboard.rb:11]: I don't think we need to add/show the `id` 
In a way, I find that the `id` is for the db and the routes, but it's not for the user

File: config/routes.rb (hunk 1 above)
Comment (Esseme) [reply config/routes.rb:6]: This specific `root` is for the `admin user` only.

When the `admin user` logs in, for now, they are routed to the index of all the articles in the app.

Comment (gkosmo) [reply config/routes.rb:6]: Cool cool
it makes sense :+1:

File: app/dashboards/article_dashboard.rb (hunk 2 above)
Comment (Esseme) [reply app/dashboards/article_dashboard.rb]: I have published some new changes to remove the `id`.

**Review by purinkle — COMMENTED** [review]
:family_woman_boy:

There is a lot more code in this change than I expected. I looked at your description of the pull request.

> We have added an admin dashboard so that the admin user can easily moderate the site.

Is the intention of this change to add a full moderation suite? Instead, how would you feel about adding a more minor change? What would you think about something simpler?

In my mind, I imagined you would add a boolean `admin` field to your `User` model. You could then add some logic around showing the "Edit" button. You would also need to add some code to your controller, which would be much less than what you have here.

Another advantage is that testing this more lightweight solution is much easier.

Does what I said above make sense? What would you change?

**Pull request merged**